### PR TITLE
Protect IgnoresAccessChecksTo, Add Task Scheduler Context, Lock on Console Log Flushing

### PR DESCRIPTION
**Pull request by Tyler-IN**
Move Impl of IgnoreAccessChecksToAttribute to Robust.Shared, make it internal

launch server into task scheduler context with custom task scheduler that does not run things inline / on the main thread

### CODE REVIEW

**Comment by PJB3005** [comment]
Please fix the compiler warning for `_offThread` being unused now.

File: Robust.Client/GameController.cs [hunk 1]
```diff
@@ -331,6 +331,10 @@ internal static void SetupLogging(ILogManager logManager, Func<ILogHandler> logH
                 args.SetObserved(); // don't crash
 #endif
             };
+
+#if !DEBUG
+            ConsoleLogHandler.TryDetachFromConsoleWindow();
```
Comment (PJB3005) [inline Robust.Client/GameController.cs:336]: Wait hang on I'm re-reading this. What's this supposed to accomplish?

File: Robust.Shared/Asynchronous/RobustTaskScheduler.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            if ((options & TaskCreationOptions.LongRunning) == 0)
+            {
+                ThreadPool.UnsafeQueueUserWorkItem((IThreadPoolWorkItem) task, false);
+            }
+            else
+            {
+                var wantsToBeFair = (task.CreationOptions & TaskCreationOptions.PreferFairness) != 0;
+                new Thread(Execute)
+                {
+                    Name = "Long Running Task",
```
Comment (PJB3005) [inline Robust.Shared/Asynchronous/RobustTaskScheduler.cs:60]: Also: I would prefer if the main game thread actually had a unique name so debugging is easier.

File: Robust.Server/Program.cs [hunk 3]
```diff
@@ -41,17 +42,15 @@ internal static void Start(string[] args, bool contentStart = false)
...
 
-            _offMainThread = new Thread(() => ParsedMain(parsed, contentStart))
-            {
-                IsBackground = false,
-                Name = "Server",
-                Priority = ThreadPriority.Highest,
-                CurrentCulture = CultureInfo.InvariantCulture,
-                CurrentUICulture = CultureInfo.InvariantCulture
-            };
+            ThreadPool.SetMinThreads(Environment.ProcessorCount * 2, Environment.ProcessorCount);
```
Comment (Tyler-IN) [inline Robust.Server/Program.cs:44]:
```suggestion

            // this sets up TaskScheduler.Current for new tasks to be scheduled off the main thread
            // the LongRunning option causes it to not be scheduled on the thread pool, so it's just a plain thread with a task context
```

**Pull request merged**